### ISSUE TITLE
Question on handler implementation

### ISSUE THREAD

**Issue by JosiahBLB** [MarcHamamji/runner.nvim#19]
Hi Marc,

Thank you for such a useful plugin. I just wanted to ask you a question about the setup, I am currently trying to set my own handler for matlab. It runs as expected when I manually enter this in the terminal: 
```sh
matlab -nojvm -nodisplay -r "run('/path/to/file')"
```

For this to work in your plugin, am I doing this correctly?
```lua
return {
  'MarcHamamji/runner.nvim',
  dependencies = {
    'nvim-telescope/telescope.nvim',
    dependencies = { 'nvim-lua/plenary.nvim' },
  },
  config = function()
    require('runner').setup {}

    -- matlab handler:
    local helpers = require 'runner.handlers.helpers'
    require('runner').set_handler('matlab', function(buffer)
      helpers.shell_handler('matlab -nojvm -nodisplay -r "run(' .. vim.fn.expand '%' (buffer) .. ')"')
    end)
    
  end,
}
```

Sorry if this is the incorrect way of asking you, I didn't know where else to ask.

Many thanks,
Josiah

**Comment by MarcHamamji** [comment]
Hello,

Calling the `helpers.shell_handler` function only creates the handler, and returns it. The handler only being a function, you should run it again with the buffer number.
Also, I'm not sure why you have `(buffer)` placed after calling `vim.fn.expand`, you should just be able to remove it.

This should fix it:
```lua
require('runner').set_handler('matlab', function(buffer)
  helpers.shell_handler('matlab -nojvm -nodisplay -r "run(' .. vim.fn.expand '%' .. ')"')(buffer)
end)
```

**Comment by JosiahBLB** [comment]
Hey Marc thanks for the help, it works for the first call. But after closing the window and re-running I get the following error:
```
E5108: Error executing lua: ...hare/nvim/lazy/runner.nvim/lua/runner/handlers/utils.lua:12: Invalid buffer id: 11
stack traceback:
        [C]: in function 'nvim_buf_delete'
        ...hare/nvim/lazy/runner.nvim/lua/runner/handlers/utils.lua:12: in function 'create_buffer'
        ...re/nvim/lazy/runner.nvim/lua/runner/handlers/helpers.lua:38: in function 'handler'
        ...h/.local/share/nvim/lazy/runner.nvim/lua/runner/init.lua:64: in function <...h/.local/share/nvim/lazy/runner.nvim/lua/runner/init.lua:44>
```
Again, I appreciate your time :)

**Comment by MarcHamamji** [comment]
It seems like a bug. Although I can't reproduce it, I think I have found the problem, and fixed it in 8358ab7.
Can you please check if it works now?

Welcome :)

**Comment by JosiahBLB** [comment]
That fixed it, thank you for the help!
